asdf/asdftypes.py

```diff
@@ -460,7 +460,7 @@ def to_tree_tagged(cls, node, ctx):
         `to_tree`, allows types to customize how the result is tagged.
         """
         obj = cls.to_tree(node, ctx)
-        return tagged.tag_object(cls.yaml_tag, obj)
+        return tagged.tag_object(cls.yaml_tag, obj, ctx=ctx)
```

Comment:
Isn't `obj` already containing `ctx`? Does it need to be passed separately and wouldn't this increase memory usage?